### PR TITLE
Bare bones json format check + error states

### PR DESCRIPTION
**Pull request by sidneymbell**
Goal is to ensure we have something that at least *can* be parsed as a tree -- that is, there must be at least 2 nodes (a root and 1 child), and they must both have names (per type definition of `NSNode`). 

Notes:
* This does not include the more formal tree ingestion tests (see outstanding issue)
* For now, state management is a simple boolean; this may grow into a more detailed piece of state a la `fetchData` in the future as we add more tests and/or formats. 
* Includes handling of edge case where tree ingestion fails (either via `fetch` or a bad JSON) and then the user subsequently uploads a good JSON (i.e., clear all tree-related errors once we have a good tree on board).

### CODE REVIEW

File: src/reducers/global.ts [hunk 1]
```diff
@@ -340,13 +360,22 @@ export const global = (state = defaultState, action: any) => {
...
         divisionOptions,
+        showTreeFormatError: false,
         mrcaOptions: traverse_preorder(tree).filter(
           (node: Node) => node.children.length >= 2
         ),
         cladeSliderField,
         cladeSliderValue: formatMrcaSliderOptionValue(tree, cladeSliderField),
         mrca: tree,
         metadataCensus: { ...state.metadataCensus, ...treeMetadata },
+        fetchData: {
```
Comment (vincent-czi) [inline src/reducers/global.ts:371]: The code as written will work, but it's probably better practice to reuse the initial default to overwrite. That way if the internal structure for `fetchData` changes down the line, this portion of the code doesn't need to know any of those implementation details. Instead, here you'd do this:
```
fetchData: {
  ...defaultState.fetchData,
},
```
This ensures it goes back to whatever the default state winds up being down the road and it's clearer to read if you aren't familiar with the `fetchData` structure since the purpose is just to reset to default.

File: src/utils/nextstrainAdapter.ts [hunk 2]
```diff
@@ -72,6 +72,33 @@ export const initializeTree = (node: NSNode, parent: Node | "root") => {
...
 
+export const validateNextstrainJson = (nextstrainJSON: NSJSON) => {
+  // top level must have at least `tree` and `meta` as objects
+  if (
+    !(
+      Object.keys(nextstrainJSON).includes("tree") &&
+      Object.keys(nextstrainJSON).includes("meta")
+    )
+  ) {
+    throw new Error();
```
Comment (vincent-czi) [inline src/utils/nextstrainAdapter.ts:83]: It's usually best practice to put a message in your Error, simply to help debugging if the Error ever escapes and starts breaking stuff. You can throw anonymous errors like this, but you'd probably be better off putting in a concise, dev-facing message.
`throw new Error("tree JSON missing expected top level key")`, etc

File: src/utils/nextstrainAdapter.ts [hunk 3]
```diff
@@ -72,6 +72,33 @@ export const initializeTree = (node: NSNode, parent: Node | "root") => {
...
+  }
+  const tree = nextstrainJSON.tree;
+  if (
+    // root node must have a name and children
+    !Object.keys(tree).includes("children") ||
+    !Object.keys(tree).includes("name")
+  ) {
+    throw new Error();
```
Comment (vincent-czi) [inline src/utils/nextstrainAdapter.ts:91]: same as above

File: src/utils/nextstrainAdapter.ts [hunk 4]
```diff
@@ -72,6 +72,33 @@ export const initializeTree = (node: NSNode, parent: Node | "root") => {
...
+  }
+  if (
+    // all of the root node's children must have a name (might be tips which in nextstrain's schema do not have a `children` attribute)
+    //@ts-ignore
+    !tree.children.every((ch: NSNode) => Object.keys(ch).includes("name"))
+  ) {
+    throw new Error();
```
Comment (vincent-czi) [inline src/utils/nextstrainAdapter.ts:98]: same as above

**Review by vincent-czi — APPROVED** [review]
Overall looks good. Comments would be nice to address, but would also be fine to merge now and just try to remember to improve those lines on some future PR if you'd rather.

File: src/components/header/invalidJsonBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          dispatch({ type: ACTION_TYPES.CLEAR_TREE_FORMAT_ERROR });
+        }}
+      >
+        <AlertTitle>
+          <strong>Woops! We can't read that tree file</strong>
+        </AlertTitle>
+        <span>
+          We weren't able to parse your tree JSON file. If you believe this is
+          in error, please{" "}
+          <a href="mailto:galago@chanzuckerberg.com">get in touch</a> so we can
```
[the remark this answers is not in the document]
Comment (sidneymbell) [reply src/components/header/invalidJsonBanner.tsx:31]: Yup!

File: src/reducers/global.ts (hunk 1 above)
Comment (sidneymbell) [reply src/reducers/global.ts:371]: Oooh good call, thanks!

File: src/utils/nextstrainAdapter.ts (hunk 2 above)
Comment (sidneymbell) [reply src/utils/nextstrainAdapter.ts:83]: Fair. Done!

**Comment by sidneymbell** [comment]
Thanks for your comments, @vincent-czi ! Implemented. Will push then merge momentarily.
